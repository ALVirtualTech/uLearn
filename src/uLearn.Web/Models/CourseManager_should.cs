using System;
using System.Collections.Generic;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace uLearn.Web.Models
{
	[TestFixture]
	public class CourseManager_should
	{
		[Test, UseReporter(typeof(DiffReporter))]
		public void load_all_courses()
		{
			var manager = WebCourseManager.Instance;
			var courses = manager.GetCourses().ToList();
			foreach (var c in courses)
			{
				Console.WriteLine("Course: " + c.Id + " " + c.Title);
				foreach (var s in c.Slides)
					Console.WriteLine(s.Info.UnitName + " " + s.Title);
				Console.WriteLine();
			}
			Approvals.VerifyAll(courses.Where(c => !c.Id.StartsWith("BasicProgramming")), "Courses");
		}

		[Test, UseReporter(typeof(DiffReporter))]
		public void load_ForTestsCourse_slides()
		{
			var manager = WebCourseManager.Instance;
			Approvals.VerifyAll(manager.GetCourse("ForTests").Slides, "Slides");
			
		}
	}
}
