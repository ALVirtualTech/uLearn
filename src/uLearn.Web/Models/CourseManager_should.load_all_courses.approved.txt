Courses[0] = Id: AIML, Title: Искусственный интеллект и машинное обучение
Courses[1] = Id: ForTests, Title: Шаблонный курс
Courses[2] = Id: Help, Title: Справка для авторов курсов (xml)
Courses[3] = Id: Linq, Title: Практикум по LINQ
Courses[4] = Id: OOP, Title: Практикум по объектно-ориентированному программированию
